dotCMS/src/main/java/com/dotmarketing/db/DotCMSInitDb.java

```diff
@@ -1,126 +1,113 @@
-		}
-	}
+        if (!isConfigured()) {
 
-	private static void loadStarterSite(PrintWriter pw) throws IOException{
-		
-		String starter = Config.getStringProperty("STARTER_DATA_LOAD");
-		File starterZip = null;
-		
-		if(UtilMethods.isSet(starter)){
+            Logger.info(DotCMSInitDb.class, "There are no inodes - initializing db with starter site");
+
+    
+            
+            Try.run(() -> loadStarterSite()).getOrElseThrow(e->new DotRuntimeException(e));
+
+
+        } else {
+            Logger.info(DotCMSInitDb.class, "inodes exist, skipping initialization of db");
+        }
+    }
+
+    
+    @WrapInTransaction
+    private static void loadStarterSiteData() throws Exception{
+        String starter = Config.getStringProperty("STARTER_DATA_LOAD", null);
+        File starterZip = null;
+        
+        if(UtilMethods.isSet(starter)){
 
-			// First we try using the real path
+            // First we try using the real path
             starterZip = new File(FileUtil.getRealPath(starter));
 
             // Then we try to see if there is an absolute path (or relative in case of integration tests)
             if (!starterZip.exists()) {
-				starterZip = new File(starter);
-			}
-		}
+                starterZip = new File(starter);
+            }
+        }
+        
+        if(starterZip==null || (starterZip!=null && !starterZip.exists())){
+            String starterSitePath = "/starter.zip";
+            String zipPath = FileUtil.getRealPath(starterSitePath);
+            starterZip = new File(zipPath); 
+         }
+        
+        ImportStarterUtil ieu = new ImportStarterUtil(starterZip);
+
+        ieu.doImport();
+
+    }
+    @CloseDBIfOpened
+	private static void loadStarterSite() throws Exception{
+		
+	    loadStarterSiteData() ;
+	    
+		DbConnectionFactory.closeAndCommit();
 		
-		if(starterZip==null || (starterZip!=null && !starterZip.exists())){
-			String starterSitePath = "/starter.zip";
-			String zipPath = FileUtil.getRealPath(starterSitePath);
-			starterZip = new File(zipPath); 
-		 }
+
+        MaintenanceUtil.flushCache();
+        ReindexThread.startThread();
+
+        ContentletAPI conAPI = APILocator.getContentletAPI();
+        Logger.info(DotCMSInitDb.class, "Building Initial Index");
+
+
+        // Initializing felix
+        OSGIUtil.getInstance().initializeFramework();
+
+        // Reindexing the recently added content
+        conAPI.refreshAllContent();
+        long recordsToIndex = APILocator.getReindexQueueAPI().recordsInQueue();
+        Logger.info(DotCMSInitDb.class, "Records left to index : " + recordsToIndex);
+
+        int counter = 0;
+
+        while (recordsToIndex > 0) {
+            Thread.sleep(2000);
+            recordsToIndex = APILocator.getReindexQueueAPI().recordsInQueue();
+            Logger.info(DotCMSInitDb.class, "Records left to index : " + recordsToIndex);
+            // ten minutes
+            if(++counter>30000) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using Literals in Conditional Statements](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)